core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/RecipesV4Controller.java

```diff
@@ -67,33 +81,38 @@ public RecipeV4Response post(Long workspaceId, RecipeV4Request request) {
     }
 
     @Override
-    public RecipeV4Response deleteByName(Long workspaceId, String name) {
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.DELETE_RECIPE)
+    public RecipeV4Response deleteByName(Long workspaceId, @ResourceName String name) {
         Recipe deleted = recipeService.delete(NameOrCrn.ofName(name), workspaceId);
         notify(ResourceEvent.RECIPE_DELETED);
         return converterUtil.convert(deleted, RecipeV4Response.class);
     }
 
     @Override
-    public RecipeV4Response deleteByCrn(Long workspaceId, @NotNull String crn) {
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.DELETE_RECIPE)
+    public RecipeV4Response deleteByCrn(Long workspaceId, @NotNull @ResourceCrn String crn) {
         Recipe deleted = recipeService.delete(NameOrCrn.ofCrn(crn), workspaceId);
         notify(ResourceEvent.RECIPE_DELETED);
         return converterUtil.convert(deleted, RecipeV4Response.class);
     }
 
     @Override
-    public RecipeV4Responses deleteMultiple(Long workspaceId, Set<String> names) {
+    @CheckPermissionByResourceNameList(action = AuthorizationResourceAction.DELETE_RECIPE)
+    public RecipeV4Responses deleteMultiple(Long workspaceId, @ResourceNameList Set<String> names) {
         Set<Recipe> deleted = recipeService.deleteMultipleByNameFromWorkspace(names, workspaceId);
         notify(ResourceEvent.RECIPE_DELETED);
         return new RecipeV4Responses(converterUtil.convertAllAsSet(deleted, RecipeV4Response.class));
     }
 
     @Override
+    @DisableCheckPermissions
```

Comment:
We should remove this endpoint?
Can we do a lookup in kibana if it's still used or not? There should be a controller interceptor for it.

Reply:
I am not sure, I haven't checked, I just think that this endpoint is not a security risk anyway

Reply:
Well, you can read anybody's recipe and it can contains sensitive information.

Reply:
yes, you're right, fixed